include/bluetooth/mesh/gen_battery_cli.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_gen_battery_cli Generic Battery Client model
+ * @{
+ * @brief API for the Generic Battery Client model.
+ */
+
+#ifndef BT_MESH_GEN_BATTERY_CLI_H__
+#define BT_MESH_GEN_BATTERY_CLI_H__
+
+#include <bluetooth/mesh/gen_battery.h>
+#include <bluetooth/mesh/model_types.h>
+
+struct gen_battery_cli;
+
+/** @def GEN_BATTERY_CLI
+ *
+ * @brief Initialization parameters for a @ref gen_battery_cli instance.
+ *
+ * @param[in] _status_handler Optional status message handler.
+ */
+#define GEN_BATTERY_CLI(_status_handler)                                       \
+	{                                                                      \
+		.status_handler = _status_handler,                             \
+		.pub = {.msg = NET_BUF_SIMPLE(2 + 0 + 4) }                     \
+	}
+
+/** @def BT_MESH_MODEL_GEN_BATTERY_CLI
+ *
+ * @brief Generic Battery Client model composition data entry.
+ *
+ * @param[in] _cli Pointer to a @ref gen_battery_cli instance.
+ */
+#define BT_MESH_MODEL_GEN_BATTERY_CLI(_cli)                                    \
+	BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_GEN_BATTERY_CLI,                     \
+			 _gen_battery_cli_op, &(_cli)->pub,                    \
+			 MODEL_USER_DATA(struct gen_battery_cli, _cli),        \
+			 &_gen_battery_cli_cb)
+
+/**
+ * Generic Battery Client structure.
+ *
+ * Should be initialized with the @ref GEN_BATTERY_CLI macro.
+ */
+struct gen_battery_cli {
+	/** @brief Asynchronous status message handler.
+	 *
+	 * Optional status message callback for handling asynchronous status
+	 * messages.
+	 *
+	 * @note This handler will not be called on messages that are sent as a
```

Comment:
Yeah, I was a bit back and forth on whether to do the status callback for acked messages. The status opcode handlers are made a bit messy because of this, so changing it would clean the module a bit as well.